java/code/src/com/suse/manager/utils/SaltUtils.java

```diff
@@ -652,7 +652,8 @@ else if (action.getActionType().equals(ActionFactory.TYPE_HARDWARE_REFRESH_LIST)
         else if (action.getActionType().equals(ActionFactory.TYPE_DIST_UPGRADE)) {
             DistUpgradeAction dupAction = (DistUpgradeAction) action;
             DistUpgradeActionDetails actionDetails = dupAction.getDetails();
-            if (actionDetails.isDryRun()) {
+            if (serverAction.getStatus().equals(ActionFactory.STATUS_FAILED) || actionDetails.isDryRun()) {
```

Comment:
Uhhh I think this is wrong.
There is a reason why we do not revert the channels after a failed DUP.

Reason: we do not know "when" it failed. In case the migration started already and you abort becuase of "disk full" or similar error, you have no chance to fix it and continue the migration.

Reply:
> Reason: we do not know "when" it failed. In case the migration started already and you abort becuase of "disk full" or similar error, you have no chance to fix it and continue the migration.

In that case, does it even matter if the channels are reverted or not?

Reply:
Ok, then let me revert this one anyway.
Maybe we can further inspect the result and revert only in case of some exceptional cases. But that's a bigger thing for later.